pkg/reconciler/pingsource/jobrunner/runner.go

```diff
@@ -89,17 +87,12 @@ func (a *cronJobsRunner) Start(stopCh <-chan struct{}) error {
 	return nil
 }
 
-func (a *cronJobsRunner) cronTick(ctx context.Context, event cloudevents.Event, reportArgs source.ReportArgs) func() {
+func (a *cronJobsRunner) cronTick(ctx context.Context, event cloudevents.Event) func() {
 	return func() {
-		// Send event (cannot be interrupted)
-		rctx, _, err := a.Client.Send(ctx, event)
-		rtctx := cloudevents.HTTPTransportContextFrom(rctx)
-		if err != nil {
-			// TODO: retries, dls
-			a.Logger.Error("failed to send cloudevent", zap.Error(err))
+		if result := a.Client.Send(ctx, event); cloudevents.IsNACK(result) {
```

Comment:
result could be ack/nack or another generic error (https://github.com/cloudevents/sdk-go#status), so maybe the correct condition want to test here is `!cloudevents.IsACK(result)`

Reply:
Thanks! Fixed!